src/api/routes/rosetta/construction.ts

```diff
@@ -297,5 +324,174 @@ export function createRosettaConstructionRouter(db: DataStore): RouterWithAsync
+        recipient: recipientAddress,
+        amount: new BN(amount),
+        fee: new BN(fees),
+        publicKey: publicKeys[0].hex_bytes,
+        network: GetStacksTestnetNetwork(),
+        nonce: accountInfo.nonce ? new BN(accountInfo.nonce) : new BN(0),
+      };
+    }
+
+    const transaction = await makeUnsignedSTXTokenTransfer(tokenTransferOptions);
+    const unsignedTransaction = transaction.serialize();
+    const hexBytes = digestSha512_256(unsignedTransaction).toString('hex');
+    const response: RosettaConstructionPayloadResponse = {
+      unsigned_transaction: unsignedTransaction.toString('hex'),
+      payloads: [
+        {
+          address: senderAddress,
+          hex_bytes: '0x' + hexBytes,
+          signature_type: 'ecdsa',
+        },
+      ],
+    };
+    res.json(response);
+  });
+
+  //construction/combine endpoint
+  router.postAsync('/combine', async (req, res) => {
+    const valid: ValidSchema = await rosettaValidateRequest(req.originalUrl, req.body);
+    if (!valid.valid) {
+      res.status(400).json(makeRosettaError(valid));
+      return;
+    }
+
+    const combineRequest: RosettaConstructionCombineRequest = req.body;
+    const signatures = combineRequest.signatures;
+
+    if (has0xPrefix(combineRequest.unsigned_transaction)) {
+      res.status(400).json(RosettaErrors.invalidTransactionString);
+      return;
+    }
+
+    if (signatures.length === 0) {
+      res.status(400).json(RosettaErrors.noSignatures);
+      return;
+    }
+
+    let unsigned_transaction_buffer: Buffer;
+    let transaction: StacksTransaction;
+
+    try {
+      unsigned_transaction_buffer = hexToBuffer('0x' + combineRequest.unsigned_transaction);
+      transaction = deserializeTransaction(BufferReader.fromBuffer(unsigned_transaction_buffer));
+    } catch (e) {
+      res.status(400).json(RosettaErrors.invalidTransactionString);
+      return;
+    }
+
+    for (const signature of signatures) {
+      if (signature.public_key.curve_type !== 'secp256k1') {
+        res.status(400).json(RosettaErrors.invalidCurveType);
+        return;
+      }
+      const preSignHash = makePresignHash(transaction);
+      if (!preSignHash) {
+        res.status(400).json(RosettaErrors.invalidTransactionString);
+        return;
+      }
+
+      let newSignature: MessageSignature;
+
+      try {
+        newSignature = createMessageSignature(signature.signing_payload.hex_bytes);
+      } catch (error) {
+        res.status(400).json(RosettaErrors.invalidSignature);
+        return;
+      }
+
+      if (!verifySignature(preSignHash, signature.public_key.hex_bytes, newSignature)) {
+        res.status(400).json(RosettaErrors.signatureNotVerified);
+      }
+
+      if (transaction.auth.spendingCondition && isSingleSig(transaction.auth.spendingCondition)) {
+        transaction.auth.spendingCondition.signature = newSignature;
+      } else {
+        const authField = createTransactionAuthField(newSignature);
```

Comment:
This branch is still not covered by tests

Reply:
api.ts-> Line number 1575 covers this branch.

Reply:
Check the code coverage output, the line is never it. It appears the test exits early